core/src/main/java/org/infinispan/globalstate/GlobalConfigurationManager.java

```diff
@@ -0,0 +1,42 @@
+package org.infinispan.globalstate;
+
+import java.util.EnumSet;
+
+import org.infinispan.Cache;
+import org.infinispan.commons.api.CacheContainerAdmin;
+import org.infinispan.configuration.cache.Configuration;
+import org.infinispan.factories.scopes.Scope;
+import org.infinispan.factories.scopes.Scopes;
+
+/**
+ * The {@link GlobalConfigurationManager} is the main interface for sharing runtime configuration state across a cluster.
+ * It uses an internal cache 'org.infinispan.CONFIG'. The cache is keyed with {@link ScopedState}. Each scope owner is responsible
+ * for its own keys.
+ *
+ * @author Tristan Tarrant
+ * @since 9.2
+ */
+@Scope(Scopes.GLOBAL)
+public interface GlobalConfigurationManager {
+   String CONFIG_STATE_CACHE_NAME = "org.infinispan.CONFIG";
+   /**
+    * Defines a cluster-wide cache configuration
+    *  @param cacheName the name of the configuration
+    * @param configuration the configuration object
+    * @param flags
+    */
+   Configuration createCache(String cacheName, Configuration configuration, EnumSet<CacheContainerAdmin.AdminFlag> flags);
```

Comment:
I would say to make these just `Set` instead of `EnumSet`.

Reply:
Why ? It is easier to build an EnumSet using EnumSet.of(...). Alternatively we could use varargs

Reply:
`EnumSet` implements `Set` already so it is just using the interface is all on the method.

Reply:
or a collection

Reply:
This is not public API anyway so we don't have to be picky :)

Reply:
why returning `Configuration`?